### PR TITLE
feat(paginated vaults): use flat array over nested vaults PE-7081

### PR DESCRIPTION
**Pull request by fedellen**

### CODE REVIEW

File: src/vaults.lua [hunk 1]
```diff
@@ -123,21 +123,26 @@ end
...
 	local allVaults = vaults.getVaults()
 	local vaultsArray = {}
 	local cursorField = "address" -- the cursor will be the wallet address
-	for address, vault in pairs(allVaults) do
-		table.insert(vaultsArray, {
-			address = address,
-			vault = vault,
-		})
+
+	for address, vaultsForAddress in pairs(allVaults) do
```
Comment (arielmelendez) [inline src/vaults.lua:133]: Nit: you could use utils.reduce to convey the canonical higher order function that's taking place here.

**Review by arielmelendez — APPROVED** [review]

File: src/vaults.lua (hunk 1 above)
Comment (dtfiedler) [reply src/vaults.lua:133]: +1 to this rec

File: src/vaults.lua [hunk 2]
```diff
@@ -123,21 +123,26 @@ end
 --- @param cursor string|nil The address to start from
 --- @param limit number Max number of results to return
 --- @param sortOrder string "asc" or "desc" sort direction
---- @return table Array of {address, vault} objects
-function vaults.getPaginatedVaults(cursor, limit, sortOrder)
+--- @param sortBy string|nil "address", "vaultId", "balance", "startTimestamp", "endTimestamp" field to sort by
+--- @return table Array of { address, vaultId, balance, startTimestamp, endTimestamp } objects
+function vaults.getPaginatedVaults(cursor, limit, sortOrder, sortBy)
 	local allVaults = vaults.getVaults()
```
[the remark this answers is not in the document]
Comment (fedellen) [reply src/vaults.lua:129]: do you mean another version of this function, or just renaming this one?

File: src/vaults.lua [hunk 3]
```diff
@@ -123,21 +123,26 @@ end
 --- @param cursor string|nil The address to start from
 --- @param limit number Max number of results to return
 --- @param sortOrder string "asc" or "desc" sort direction
---- @return table Array of {address, vault} objects
-function vaults.getPaginatedVaults(cursor, limit, sortOrder)
+--- @param sortBy string|nil "address", "vaultId", "balance", "startTimestamp", "endTimestamp" field to sort by
+--- @return table Array of { address, vaultId, balance, startTimestamp, endTimestamp } objects
```
[the remark this answers is not in the document]
Comment (fedellen) [reply src/vaults.lua]: b3cb37b

**Review by dtfiedler — APPROVED** [review]

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**